eth/downloader/downloader.go

```diff
@@ -751,7 +751,7 @@ func (d *Downloader) fillHeaderSkeleton(from uint64, skeleton []*types.Header) (
 			return d.queue.ReserveHeaders(p, count), false, false
 		}
 		fetch = func(p *peerConnection, req *fetchRequest) error {
-			return p.FetchHeaders(req.From, MaxHeaderFetch)
+			return p.FetchHeaders(req.From, int(req.From-req.To))
```

Comment:
This should still check to make sure int(req.From-req.To) <= MaxHeaderFetch. Please create an issue.

Reply:
https://github.com/dominant-strategies/go-quai/issues/671